### PR TITLE
fix(SUP-43778):v7 Player Hotspot Text Formatting

### PR DESCRIPTION
**Pull request by inbalvasserman**
issue:
*hotspots didn't break line (in kea it breaks lines)
*when change player size the hotspots text and border radius didn't changed accordingly

solution:
*change css style to fit to kea style and allow break lines
*calculate the font size and border radius relatively to player size

solved [SUP-43778](https://kaltura.atlassian.net/browse/SUP-43778)

[SUP-43778]: https://kaltura.atlassian.net/browse/SUP-43778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by SivanA-Kaltura** [comment]
@inbalvasserman Try to upgrade cypress to 13.13.1 (fixed version, no caret). This should fix the firefox timeout.

File: src/hotspots-plugin.tsx [hunk 1]
```diff
@@ -105,6 +112,14 @@ export class HotspotsPlugin extends KalturaPlayer.core.BasePlugin {
     };
   }
 
+  private _calculateStyle(cuepoint: LayoutHotspot, scaleCalculation: ScaleCalculation): Style {
+    const {rawLayout, styles} = cuepoint;
+    return {
+      fontSize: parseInt(styles['font-size'])/ rawLayout.stageWidth * scaleCalculation.width,
```
Comment (SivanA-Kaltura) [inline src/hotspots-plugin.tsx:118]: are these values always set ?

File: src/components/Hotspot.tsx [hunk 2]
```diff
@@ -217,14 +182,13 @@ export default class Hotspot extends Component<Props, State> {
...
 
-    const fontSizeToUse = `${this.getFontSize(layout, hotspot, label || '')}px`;
-
     const buttonStyles = {
       ...defaultButtonsStyles,
       ...hotspot.styles,
       cursor: disableClick ? 'default' : 'pointer',
       maxWidth: `${layout.width}px`,
-      fontSize: fontSizeToUse
+      fontSize: `${hotspot.relativeStyle.fontSize}px`,
```
Comment (SivanA-Kaltura) [inline src/components/Hotspot.tsx:190]: are these values always set ?

File: src/components/Hotspot.tsx [hunk 3]
```diff
@@ -11,20 +11,21 @@ const defaultContainerStyles = {
 };
 
 const defaultButtonsStyles = {
-  position: 'relative',
+  position: 'absolute',
```
Comment (SivanA-Kaltura) [inline src/components/Hotspot.tsx:14]: this would position the button according to the first position ancestor. which problem does this change fix ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/components/Hotspot.tsx (hunk 2 above)
Comment (inbalvasserman) [reply src/components/Hotspot.tsx:190]: this is build in the code so yes

File: src/hotspots-plugin.tsx (hunk 1 above)
Comment (inbalvasserman) [reply src/hotspots-plugin.tsx:118]: yes, kea sent the data like this

Comment (SivanA-Kaltura) [reply src/hotspots-plugin.tsx:118]: what about kms ? it has a different editor i think

File: src/components/Hotspot.tsx (hunk 3 above)
Comment (SivanA-Kaltura) [reply src/components/Hotspot.tsx:14]: mmmm....
i don't know what the design here is, but this sounds like a case where a tooltip should be used

File: src/components/Hotspot.tsx (hunk 2 above)
Comment (SivanA-Kaltura) [reply src/components/Hotspot.tsx:190]: i mean
layout.width is always set ?
hotspot.relativeStyle.fontSize is always set ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

**Review by semarche-kaltura — APPROVED** [review]

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**